logstash/check.py

```diff
@@ -0,0 +1,304 @@
+                headers=headers(self.agentConfig),
+                auth=auth,
+                verify=verify,
+                cert=cert
+            )
+            resp.raise_for_status()
+        except Exception as e:
+            if send_sc:
+                self.service_check(
+                    self.SERVICE_CHECK_CONNECT_NAME,
+                    AgentCheck.CRITICAL,
+                    message="Error {0} when hitting {1}".format(e, url),
+                    tags=config.service_check_tags
+                )
+            raise
+
+        return resp.json()
+
+    def _get_logstash_version(self, config):
+        """ Get the running version of logstash.
+        """
+        try:
+            data = self._get_data(config.url, config, send_sc=False)
+            version = data['version']
+            version = map(int, version.split('.')[0:3])
+        except Exception as e:
+            self.warning(
+                "Error while trying to get Logstash version "
+                "from %s %s"
+                % (config.url, str(e))
+            )
+            version = [1, 0, 0]
+
+
+        self.service_metadata('version', version)
+        self.log.debug("Logstash version is %s" % version)
+        return version
+
+    def check(self, instance):
+        config = self.get_instance_config(instance)
+
+        version = self._get_logstash_version(config)
+
+        stats_metrics = dict(self.STATS_METRICS)
+        if version < [6, 0, 0]:
+            stats_metrics.update(self.PIPELINE_METRICS)
+            pipeline_input_metrics = self.PIPELINE_INPUTS_METRICS
+            pipeline_output_metrics = self.PIPELINE_OUTPUTS_METRICS
+            pipeline_filters_metrics = self.PIPELINE_FILTERS_METRICS
+
+        stats_url = urlparse.urljoin(config.url, '/_node/stats')
+        stats_data = self._get_data(stats_url, config)
+
+        self._process_stats_data(stats_data, stats_metrics, config)
+
+        if version < [6, 0, 0]:
+            self._process_pipeline_input_data(stats_data, pipeline_input_metrics, config)
+            self._process_pipeline_output_data(stats_data, pipeline_output_metrics, config)
+            self._process_pipeline_filters_data(stats_data, pipeline_filters_metrics, config)
+
+        self.service_check(
+            self.SERVICE_CHECK_CONNECT_NAME,
+            AgentCheck.OK,
+            tags=config.service_check_tags
+        )
+
+    def _process_stats_data(self, data, stats_metrics, config):
+        for metric, desc in stats_metrics.iteritems():
+            self._process_metric(data, metric, *desc, tags=config.tags)
+
+    def _process_pipeline_input_data(self, data, pipeline_inputs_metrics, config):
+        plugins_data = data['pipeline']['plugins']
+        for inputs_data in plugins_data.get('inputs',[]):
+            inputs_name = inputs_data.get('name')
+
+            metrics_tags = list(config.tags)
+
+            if inputs_name:
+                metrics_tags.append(
```

Comment:
Would it be useful to tag input data with `input_name:unknown" if they don't have a "name" (that way the tag could be use in graph and alerts) ?